src/Lean/Data/AssocList.lean

```diff
@@ -54,17 +54,34 @@ def mapVal (f : β → δ) : AssocList α β → AssocList α δ
   | nil        => nil
   | cons k v t => cons k (f v) (mapVal f t)
 
-def findEntry? [BEq α] (a : α) : AssocList α β → Option (α × β)
+/-- Auxiliary for `List.reverse`. `List.reverseAux l r = l.reverse ++ r`, but it is defined directly. -/
+def reverseAux : AssocList α β → AssocList α β → AssocList α β
+  | nil,   r => r
+  | cons k v l, r => reverseAux l (cons k v r)
+
+def reverse (as : AssocList α β) : AssocList α β := reverseAux as nil
+
+@[inline]
+def mapValM {m : Type u → Type v} [Monad m] {α : Type u} {β γ : Type _} (f : β → m γ) (as : AssocList α β) : m (AssocList α γ) :=
+  let rec @[specialize] loop
+    | nil,         bs => pure bs.reverse
+    | cons k v as, bs => do loop as (cons k (← f v) bs)
+  loop as nil
+
+def getEntry? [BEq α] (a : α) : AssocList α β → Option (α × β)
   | nil         => none
   | cons k v es => match k == a with
     | true  => some (k, v)
-    | false => findEntry? a es
+    | false => getEntry? a es
 
-def find? [BEq α] (a : α) : AssocList α β → Option β
+def get? [BEq α] (a : α) : AssocList α β → Option β
```

Comment:
I don't like this naming because it sounds too much like a function that returns the n-th element of the list. `find` has more of a search/lookup connotation.

Reply:
@digama0 one problem with using `find` for this distinction is that is already used for e.g. `Array` to denote the operation that searches for the element matching a predicate, not one indexes by key.

Reply:
Isn't that one called `findP`?

Reply:
@digama0 no, they are [`List.find?`](https://github.com/leanprover/lean4/blob/0b01ceb3bb3334fd6c2f3b24996646046722c62a/src/Init/Data/List/Basic.lean#L337) and [`Array.find?`](https://github.com/leanprover/lean4/blob/0b01ceb3bb3334fd6c2f3b24996646046722c62a/src/Init/Data/Array/Basic.lean#L436), respectively. The convention of a search by function is carried through to `findSome?`, `findRev?`, `findIdx?` as well. Conversely, the reverse lookup of an index by value for an array is called [`Array.getIdx?`](https://github.com/leanprover/lean4/blob/0b01ceb3bb3334fd6c2f3b24996646046722c62a/src/Init/Data/Array/Basic.lean#L478).